src/auth/auth-cross-domain-iframe.ts

```diff
@@ -0,0 +1,60 @@
+//#region imports
+
+import {
+  BBAuthDomUtility
+} from '../shared/dom-utility';
+import { TokenResponse } from './token-response';
+
+//#endregion
+
+function bindEvent(element: any, eventName: string, eventHandler: any) {
+  if (element.addEventListener) {
+      element.addEventListener(eventName, eventHandler, false);
+  } else if (element.attachEvent) {
+      element.attachEvent('on' + eventName, eventHandler);
+  }
+}
+
+// URL to get iframe
+const URL = '';
+
+export class BBAuthCrossDomainIframe {
+
+  public static GetToken(): Promise<TokenResponse> {
+    let iframeEl: HTMLIFrameElement;
+    iframeEl = this.getOrMakeIframe();
+    return this.getTokenFromIFrame(iframeEl);
+  }
+
+  public static getOrMakeIframe(): HTMLIFrameElement {
+    let iframeEl = document.getElementById('auth-cross-domain-iframe') as HTMLIFrameElement;
+    // if iframe doesn't exist, make it
+    if (iframeEl === undefined || iframeEl === null) {
+      iframeEl = BBAuthDomUtility.addIframe(
+        URL,
+        'auth-cross-domain-iframe',
+        ''
+      );
+    }
+    return iframeEl;
+  }
+
+  public static getTokenFromIFrame(iframeEl: HTMLIFrameElement): Promise<TokenResponse> {
+    return new Promise<TokenResponse>((resolve) => {
+      bindEvent(window, 'message', function handleMessageFromIFrame(msg: any) {
+        if (msg.data['methodName'] === 'ready') {
```

Comment:
Bracket notation is not needed here since `methodName` has no illegal characters.  Use `msg.data.methodName` instead.